pbm/pbm_util.go

```diff
@@ -64,6 +65,7 @@ func CreateCapabilityProfileSpec(pbmCreateSpec CapabilityProfileCreateSpec) (*ty
 			SubProfiles: []types.PbmCapabilitySubProfile{
 				types.PbmCapabilitySubProfile{
 					Capability: capabilities,
+					Name:       pbmCreateSpec.SubProfileName,
```

Comment:
Since SubProfileName need not be mandatory, I think we should give the users the flexibility to create a profile without sub profile name too. 

Also, With you current code client_test.go will fail - https://github.com/vmware/govmomi/blob/master/pbm/client_test.go#L198 as sub profile name is not populated.

So, add a SubProfiles.Name to pbmCapabilityProfileSpec if CapabilityProfileCreateSpec.SubProfileName name is non-empty.

Please make sure to run the pbm tests and add it to your testing done.

Reply:
Looks like it is still optional with this change. If `CapabilityProfileCreateSpec.SubProfileName` is empty, then `PbmCapabilitySubProfile.Name` will be empty too, right?

Reply:
Aah, got it! Makes sense. 
In that case, @sumitAgrawal007 please run the PBM client tests and post in test description.